add action creator and reducer tests

MichaelRoytman opened the pull request.

MichaelRoytman commented on src/data/actions/assets.js:
```diff
@@ -41,7 +46,7 @@ export const getAssets = (request, courseDetails) =>
       })
       .catch((error) => {
-        dispatch(assetDeleteFailure(error));
+        dispatch(requestAssetsFailure(error));
```
I have changed this to be a requestAssetsFailure, since it's a failure to request assets and not a failure to delete an asset.

MichaelRoytman commented on src/data/actions/assets.js:
```diff
@@ -70,10 +75,15 @@ export const deleteAsset = (assetId, courseDetails) =>
-          dispatch(deleteAssetSuccess(assetId, response));
-        } else {
-          dispatch(assetDeleteFailure(response));
+          return response.json();
```
I have changed the response to be JSON.

fysheets replied:
Question here -- if the response from the API to delete the asset comes back with a non 2XX response, does it return any additional info as to why it failed to delete in the body? i.e could we still use a `response.json()` to note a more verbose error message to the user in the future? Or do all non-2XX responses return an empty body?

MichaelRoytman replied:
The API only returns 204 or 404, neither of which have bodies, so I have actually removed the call to jsonify the response. Does that seem like the right call?

MichaelRoytman commented on src/data/actions/assets.js:
```diff
@@ -95,16 +105,20 @@ export const toggleLockAssetFailure = (asset, response) => ({
   (dispatch) => {
     dispatch(togglingLockAsset(asset));
-    clientApi.requestToggleLockAsset(courseDetails.id, asset)
+    return clientApi.requestToggleLockAsset(courseDetails.id, asset)
```
We have to return something.

MichaelRoytman commented on src/data/actions/assets.js:
```diff
@@ -95,16 +105,20 @@ export const toggleLockAssetFailure = (asset, response) => ({
-export const clearAssetsStatus = () =>
-  dispatch =>
-    dispatch({ type: assetActions.CLEAR_ASSETS_STATUS });
+export const clearAssetsStatus = () => ({
```
I don't think this should be an async action creator, so I have changed it to be synchronous.

MichaelRoytman commented on src/data/actions/assets.test.jsx:
```diff
@@ -24,18 +24,23 @@ const middlewares = [thunk];
   it('returns expected state from assetDeleteFailure', () => {
     const expectedAction = { response: 'response', type: assetActions.DELETE_ASSET_FAILURE };
-    expect(store.dispatch(actionCreators.assetDeleteFailure('response'))).toEqual(expectedAction);
+    expect(store.dispatch(actionCreators.deleteAssetFailure('response'))).toEqual(expectedAction);
```
I just re-ordered the words to be more consistent with other action creators.

MichaelRoytman commented on src/data/reducers/assets.js:
```diff
@@ -6,31 +6,31 @@ import { assetLoading } from '../constants/loadingTypes';
 import { getAssetAPIAttributeFromDatabaseAttribute } from '../../utils/getAssetsAttributes';
 
-const filtersInitial = {
+export const filtersInitial = {
```
I'm exporting all of the reducers so that I can import and test them.

MichaelRoytman commented on src/data/reducers/assets.js:
```diff
@@ -87,8 +87,18 @@ const sort = (state = sortInitial, action) => {
-const status = (state = {}, action) => {
+export const status = (state = {}, action) => {
   switch (action.type) {
+    case assetActions.REQUEST_ASSETS_SUCCESS:
```
I'm updating the status state based on the response of the request.

MichaelRoytman commented on src/data/actions/accessibility.test.jsx:
```diff
@@ -16,19 +16,50 @@ const middlewares = [thunk];
-describe('Accessibility Action Creator', () => {
+describe('Accessibility Action Creators', () => {
+  beforeEach(() => {
+    store = mockStore(initialState);
```
I'm pulling this out into a beforeEach, since each test case shares the same initial mock store state.

MichaelRoytman commented on src/data/actions/assets.js:
```diff
@@ -15,7 +15,12 @@ export const requestAssetsSuccess = response => ({
 });
 
-export const assetDeleteFailure = response => ({
+export const requestAssetsFailure = response => ({
```
I'm adding a request assets failure action, since we didn't have one before.

MichaelRoytman commented on src/data/reducers/assets.test.jsx:
```diff
@@ -0,0 +1,290 @@
+let defaultState;
+
+describe('Assets', () => {
+  describe('assets reducer', () => {
```
This refers to the assets state. I know it looks like it's redundant, but it's the assets reducer of the assets.js file.

MichaelRoytman commented:
@edx/educator-dahlia This is ready for a review! Thanks!

MichaelRoytman commented:
@fysheets I believe I have addressed the issue you brought up. I have removed call to jsonify the response for deleting an asset, since the response will be either 204 or 404, neither of which have bodies. It sounds like, from the research I have done, the failure of the json call is intentional, so we should just avoid calling json on responses without bodies.

fysheets commented:
@MichaelRoytman got it -- and sure, I'll check it out again thanks for the investigation!
